### DISCUSSION TITLE
[Feature] Hard split libraries

### DISCUSSION THREAD

**Discussion by Atakku** [immich-app/immich#3150] [Feature Request]
### The feature

I guess this would require a bit of modification to the web client uploads as they are ambiguous.
A Workaround can be achieved with having multiple users with shared albums too I guess.

But it would be really nice to have some hard separation between image libraries, like photos and pictures (downloaded/screenshots). I really like the CLIP imag searching, and there are so many occasions
I would love to use it for things other than my camera pictures, but I do not want to mix the libraries together.

### Platform

- [X] Server
- [X] Web
- [X] Mobile

**Comment by Atakku** [comment]
I'm going to try working around this by building a cloned instance of immich mobile app and a secondary immich server, as this would be more flexible than having two users with shared albums

**Comment by SmoochSmackers** [comment]
I would absolutely love something like this! I just did a bunch of work to try and get my photos from google back and separate out the downloaded media/screenshots from the photos/videos I took myself (see below if you're interested in that boring journey). I did all this with the thought that I would do the same for the media that I currently have on Immich, combine the two sets, and then put them all back into Immich in separate "Photos" and "Downloaded" directories, such that I wouldn't have the downloaded stuff clogging up my main library feed. It seems as though this isn't (yet?) a supported use case for Immich. 

### My Google Photos (almost) to Immich Journey

I recently used Google Takeout to get my Google Photos library back. This includes both my photos/videos and all my downloaded pictures/screenshots. I was able to use [GooglePhotosTakeoutHelper](https://github.com/TheLastGimbus/GooglePhotosTakeoutHelper) in conjunction with [EXIFTool ](https://exiftool.sourceforge.net/) to better extract the zip files and to ensure EXIF was preserved. I chose the "YYYY/MM" directory structure. 

Then, with the help of ChatGPT, I generated a PowerShell tool that checks each image's EXIF data for the make/model of the camera and separates them into two parent directories "Photos" (images/videos I've taken) and "Downloaded" (images/videos I've downloaded and screenshots) where the YYYY/MM directory structure is retained within each folder. Most importantly, if an image/video is slated for the "Downloaded" folder, it first cross references said file with the corresponding JSON file in the "takeout" folder left over from use by GooglePhotosTakeoutHelper. It checks whether the "googlePhotosOrigin", in said JSON, contains "photosDesktopUploader". If so, it overturns its initial decision and puts it in "Photos" instead, otherwise, it's off to "Downloads" it goes. This was important for me because some of my older photos didn't seem to have the correct EXIF data, but most files that I would have uploaded via the Desktop Uploader would have been those I took myself. IDK if this is too specific to my needs, but if anyone reading this is interested in the described powershell tool, let me know and I can try to clean it up and post it somewhere.

**Comment by kuzmandi** [comment]
Has anything progressed in this matter?

I arrived here with the same request, not knowing that there are other people wanting this.

#13866

Or what is the best workaround method?

Maybe with external libraries (mount/unmount to select which "channel(s)" I want to see? Can I deactivate the removal of meta data after 30 days?

**Comment by Shrinks99** [comment]
To add another user story to this discussion:

I've recently inherited some digital photo libraries from relatives.  I would love to store these in Immich, but separated from my personal photos, I did not take them and they aren't really _mine_, but I want to keep a copy to remember my family!  Having multiple libraries available within a single user account would help me achieve the separation I'm looking for.

**Comment by Vini-Frei** [comment]
+1
Separating libraries would be so nice for family members on the same server. Running multiple Immich instances achieves this but it can be tricky for people with low-end hardware.
